grove/grove.go

```diff
@@ -68,5 +83,33 @@ func NewWithFS(fs FS) (*Grove, error) {
 	if fs == nil {
 		return nil, fmt.Errorf("fs cannot be nil")
 	}
-	return &Grove{}, nil
+	return &Grove{
+		FS: fs,
+	}, nil
+}
+
+// Get searches the grove for a node with the given id. It returns the node if it was
+// found, a boolean indicating whether it was found, and an error (if there was a
+// problem searching for the node).
+func (g *Grove) Get(nodeID *fields.QualifiedHash) (forest.Node, bool, error) {
+	filename, err := nodeID.MarshalString()
+	if err != nil {
+		return nil, false, fmt.Errorf("failed determining file name for node: %w", err)
+	}
+	file, err := g.Open(filename)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, false, nil
```

Comment:
You know... I didn't realize that `errors.Is` worked that way. It's your turn to teach me something.

To validate this, I checked [the documentation](https://godoc.org/errors#Is), but I don't see this info there. However, I was able to confirm it in [this playground](https://play.golang.org/p/AH-78xGdeC3).

@aThorp96 Where did you pick up this tidbit? I will totally restructure to use this form when i get a minute.

Reply:
I had a hunch it had defined behavior when err was nil, and since the documentation didn't help I just checked the source code. I also checked the playgroud to make sure it behaved as expected.